build.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/sh
+
+set -eu
+
+is_substring()
+{
+    case "$2" in
+        *$1*)
+        return 0
+        ;;
+        *)
+        return 1
+        ;;
+    esac
+}
+
+OK=0
+CURRENT_DIR=$(pwd)
+MY_NAME=$(basename -- "$0")
+ROOTFS_ARCHIVE="rootfs.tar.xz"
+BUILD_DIRECTORY="${CURRENT_DIR}/build"
+ROOTFS_DIRECTORY="${BUILD_DIRECTORY}/rootfs"
+DEBIAN_VERSION="jessie"
+
+cleanup()
+{
+    if is_substring "${CURRENT_DIR}" "${BUILD_DIRECTORY}"
+    then
+        rm -rf "${BUILD_DIRECTORY}"
+    fi
+}
+
+prepare()
+{
+    if [ ! -d "${ROOTFS_DIRECTORY}/usr/bin" ]
+    then
+        mkdir -p "${ROOTFS_DIRECTORY}/usr/bin"
+    fi
+
+    if [ ! -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static" ]
+    then
+        cp /usr/bin/qemu-arm-static "${ROOTFS_DIRECTORY}/usr/bin/"
+    fi
+}
+
+
+bootstrap_rootfs()
+{
+    printf "Bootstrapping rootfs to %s\\n" "${ROOTFS_DIRECTORY}"
+    debootstrap \
+        --arch=armhf \
+        --variant=minbase \
+        --include=f2fs-tools,mtd-utils,busybox,udisks2,rsync \
+        "${DEBIAN_VERSION}" "${ROOTFS_DIRECTORY}" http://ftp.debian.com/debian
+}
+
+strip_rootfs()
+{
+    printf "Stripping rootfs\\n"
+    if [ -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static" ]
+    then
+        printf "Removing qemu\\n"
+        # TODO: Strip even more see issue EMP-324 Reduce rootfs size
+        rm -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static"
+    fi
+}
+
+compress_rootfs()
+{
+    if [ -f "${BUILD_DIRECTORY}/${ROOTFS_ARCHIVE}" ]
+    then
+        rm -f "${BUILD_DIRECTORY}/${ROOTFS_ARCHIVE}"
+    fi
+
+    printf "Compressing rootfs\\n"
+    tar -cJf "${BUILD_DIRECTORY}/${ROOTFS_ARCHIVE}" -C "${ROOTFS_DIRECTORY}" .
+    chown "$(id -un):$(id -gn)" "${BUILD_DIRECTORY}/${ROOTFS_ARCHIVE}"
+    printf "Created %s\\n" "${ROOTFS_ARCHIVE}"
+}
+
+usage()
+{
+cat <<-EOT
+Usage:   ./${MY_NAME} [ -h | -c ]
+-c   Cleanup, after a run you have the possibility to get the artifact and then cleanup
+-h   Print usage
+NOTE: This script should be run as root
+EOT
+}
+
+if [ "$(id -u)" != "0" ]
+then
+    printf "Make sure this script is run as root\\n"
+    usage
+    exit
+fi
+
+while getopts ":hc" options
+do
+  case "${options}" in
+    c )
+      cleanup
+      exit ${OK}
+      ;;
+    h )
+      usage
+      exit ${OK}
+      ;;
+    :)
+      printf "Option -%s requires an argument.\\n" "${OPTARG}"
+      ;;
+    \?)
+      printf "Invalid option: -%s\\n" "${OPTARG}"
+      ;;
+  esac
+done
+shift "$((OPTIND-1))"
+
+
+cleanup
```

Comment:
Yes it will cleanup "before" it builds, to start fresh. After a build we want to be able to fetch the artifact from the build directory and then cleanup after if required.